src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/velocity/mode/VelocityReversal.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of LiquidBounce (https://github.com/CCBlueX/LiquidBounce)
+ *
+ * Copyright (c) 2015 - 2025 CCBlueX
+ *
+ * LiquidBounce is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LiquidBounce is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.module.modules.combat.velocity.mode
+
+import kotlin.math.abs
+import net.ccbluex.liquidbounce.event.events.PacketEvent
+import net.ccbluex.liquidbounce.event.events.PlayerTickEvent
+import net.ccbluex.liquidbounce.event.sequenceHandler
+import net.ccbluex.liquidbounce.features.module.modules.combat.killaura.ModuleKillAura
+import net.minecraft.network.packet.s2c.play.EntityVelocityUpdateS2CPacket
+import net.minecraft.network.packet.s2c.play.ExplosionS2CPacket
+
+/**
+ * A velocity mode that reverses your velocity after a set amount of ticks.
+ * Default values bypass Vulcan (3/9/25) ~ anticheat-test.com
+ */
+internal object VelocityReversal : VelocityMode("Reversal") {
+    private val delay by int("Reversal Delay", 2, 1..5, "ticks")
+    private val xModifier by float("X Modifier", 0.5f, 0.1f..1.0f)
+    private val zModifier by float("Z Modifier", 0.5f, 0.1f..1.0f)
+    private val requiresKillaura by boolean("RequiresKillAura", true)
+
+    private var handlingVelocity = false
+    private var velocityTicks = 0
+
+    // We assume the velocity has reset. Idk of any edge cases where this logic would fail.
+    private inline val hasVelocityReset
+        get() =
+            abs(player.velocity.x) == 0.0 &&
+                abs(player.velocity.y) == 0.0 &&
+                abs(player.velocity.z) == 0.0
```

Comment:
@sqlerrorthing 

1. This should be a function. It does not reflect a property of the VelocityReversalMode, it rather answers the question whether or not a condition is met
2. The `inline` modifier should not be there. The JVM JIT will decide if it makes sense to inline this method.
3. Just use `player.velocity.lengthSquared() == 0`

```suggestion
    private fun hasVelocityReset(): Boolean {
           return player.velocity.lengthSquared() == 0
    }
```